python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1186,27 +1189,23 @@ def can_deserialize(
         )
 
     @property
-    def partitions_def(self) -> PartitionsDefinition[T_str]:
-        return self._partitions_def
+    def partitions_def(self) -> Optional[PartitionsDefinition[T_str]]:
```

Comment:
Should we get rid of this property on `PartitionsSubset`?

Reply:
I'd like to keep it so we can define general methods on `PartitionsSubset` like `__or__`, `__sub__` and `__and__`, which require constructing empty partitions subsets. In the time window case, we need to pass in the partitions def when constructing empty subsets